modules/ecs-cluster-ec2-asg/v0.0.1/main.tf

```diff
@@ -0,0 +1,308 @@
+        "Principal" : {
+          "Service" : "ssm.amazonaws.com"
+        },
+        "Action" : [
+          "kms:Encrypt*",
+          "kms:Decrypt*",
+          "kms:ReEncrypt*",
+          "kms:GenerateDataKey*",
+          "kms:Describe*"
+        ],
+        "Resource" : "*"
+      },
+    ]
+    Version = "2012-10-17"
+  })
+}
+
+resource "aws_cloudwatch_log_group" "this" {
+  #checkov:skip=CKV_AWS_338:CloudWatch log groups log retention will be set to the desired period
+  depends_on = [
+    aws_kms_key_policy.this
+  ]
+  name              = "/ECS/Cluster/${var.cluster_name}"
+  retention_in_days = var.cloudwatch_log_retention
+  kms_key_id        = aws_kms_key.this.arn
+  tags              = var.tags
+}
+
+# Create ECS cluster
+resource "aws_ecs_cluster" "this" {
+  #checkov:skip=CKV_AWS_65:Container Insights will be enabled when required
+  name = var.cluster_name
+
+  setting {
+    name  = "containerInsights"
+    value = var.container_insights
+  }
+
+  configuration {
+    execute_command_configuration {
+      kms_key_id = aws_kms_key.this.arn
+      logging    = "OVERRIDE"
+
+      log_configuration {
+        cloud_watch_encryption_enabled = true
+        cloud_watch_log_group_name     = aws_cloudwatch_log_group.this.name
+      }
+    }
+  }
+  tags = var.tags
+}
+
+# Create ECS IAM Instance Role and Policy
+resource "random_id" "code" {
+  byte_length = 4
+}
+
+resource "aws_iam_role" "ecs_instance_role" {
+  name               = "EcsInstanceRole-${var.cluster_name}-${random_id.code.hex}"
+  assume_role_policy = var.ecs_instance_role_assume_role_policy
+}
+
+resource "aws_iam_role_policy" "ecs_instance_role_policy" {
+  #checkov:skip=CKV_AWS_290:IAM policy will be set as desired
+  #checkov:skip=CKV_AWS_355:IAM policy will be set as desired
+  name   = "EcsInstanceRolePolicy-${var.cluster_name}-${random_id.code.hex}"
+  role   = aws_iam_role.ecs_instance_role.id
+  policy = var.ecs_instance_role_policy
+}
+
+resource "aws_iam_instance_profile" "ecs_instance_profile" {
+  name = "ecsInstanceProfile-${var.cluster_name}-${random_id.code.hex}"
+  role = aws_iam_role.ecs_instance_role.name
+}
+
+# Create ECS IAM Service Role and Policy
+resource "aws_iam_role" "ecs_service_role" {
+  name               = "EcsServiceRole-${var.cluster_name}-${random_id.code.hex}"
+  assume_role_policy = var.ecs_service_role_assume_role_policy
+}
+
+resource "aws_iam_role_policy" "ecs_service_role_policy" {
+  #checkov:skip=CKV_AWS_290:IAM policy will be set as desired
+  #checkov:skip=CKV_AWS_355:IAM policy will be set as desired
+  name   = "EcsServiceRolePolicy-${var.cluster_name}-${random_id.code.hex}"
+  role   = aws_iam_role.ecs_service_role.id
+  policy = var.ecs_service_role_policy
+}
+
+# Create Launch Template
+resource "aws_launch_template" "this" {
+  default_version        = 1
+  ebs_optimized          = false
+  name                   = "lt-${var.cluster_name}"
+  image_id               = data.aws_ami.ecs_ami.id
```

Comment:
maybe make this an optional parameter, since AMI's is a flexibility you want to give (`if var.ami_id? var.ami_id : data.aws_ami.ecs_ami.id`)

Reply:
True. When using a non-Amazon AMI, we will have to provide additional user_data to install stuff, though.